### ISSUE TITLE
No Bluetooth Found. Plug in a dongle to use "Bluetooth" and "No Wi-Fi adapter found" messages after update.(Pop!_OS 22.04 LTS)

### ISSUE THREAD

**Issue by siddhant-rajhans** [pop-os/pop#3237]
**Description:**
After a recent update, I am encountering two issues:

Bluetooth: When accessing the Bluetooth settings, I see a message stating "No Bluetooth Found. Plug in a dongle to use Bluetooth." This prevents me from using Bluetooth functionality on my device.
Wi-Fi: Occasionally, the Wi-Fi settings display a message stating "No Wi-Fi adapter found." This issue resolves itself after toggling the Wi-Fi button on and off.
**Distribution:**
**_`cat /etc/os-release`_**
```
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```
**`_uname -r_`**
`6.6.10-76060610-generic`

**Steps to reproduce:**

Open the Bluetooth settings.
Observe the message "No Bluetooth Found. Plug in a dongle to use Bluetooth."
Expected behavior:

Bluetooth settings should detect the internal Bluetooth hardware and allow enabling/disabling functionality.
Wi-Fi settings should consistently detect the Wi-Fi adapter without requiring toggling the button.
Other Notes:

I have tried restarting the system and checking for additional updates, following [Troubleshooting](https://support.system76.com/articles/bluetooth/) but the issues persist.
My device has built-in Bluetooth and Wi-Fi hardware.
**`sudo apt install --reinstall pulseaudio-module-bluetooth
`**
`Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 pulseaudio-module-bluetooth : Depends: pulseaudio (= 1:15.99.1+dfsg1-1ubuntu2.1)
E: Unable to correct problems, you have held broken packages.
`

Please note: While the Wi-Fi issue appears to have a temporary workaround, it might indicate a broader problem related to hardware detection after the recent update.
Also, Bluetooth is not working period.
![image](https://github.com/pop-os/pop/assets/51508945/68d43c6c-8176-4901-a6ce-a053005724b9)

**Comment by siddhant-rajhans** [comment]
## Issue Description

### No Bluetooth Found. Plug in a dongle to use "Bluetooth"
Hi everyone,

Sharing my experience fixing a frustrating combo of issues I faced with Nvidia drivers, Bluetooth, and an external monitor in Pop!_OS 22.04.
## Problem:
After upgrading to kernel version 6.6.10 in Pop!_OS 22.04, I encountered **No Bluetooth Found. Plug in a dongle to use "Bluetooth"**  issues related to the Bluetooth functionality.

### Symptoms:
- Bluetooth functionality lost (no device found error).

### Troubleshooting Steps:

#### Initial Attempts:
1. Attempted a clean installation of the Bluetooth driver  &  bluez gnome-bluetooth using: [Link](https://support.system76.com/articles/bluetooth/).
Did not work for me.

### Solution: that worked for me 

1. After updating to kernel version 6.6.10, I encountered these problems:
My Bluetooth completely disappeared - no devices found!
2. To get Bluetooth back, I switched to kernel version 6.5.4, but then...
My second monitor decided to take a vacation - it wasn't detected at all!

### The Fix:

Turns out, the solution involved a combination of driver updates and kernel versions. Here's what worked for me:

1. Clean Nvidia Driver Removal: First, I completely removed any existing Nvidia drivers with sudo apt purge '^nvidia-*'.
2. System Housekeeping: Then, I cleaned up unused dependencies and refreshed package information using sudo apt autoremove, sudo apt clean, and sudo apt update.
3. System Package Upgrades: Next, I upgraded all system packages to their latest versions with sudo apt full-upgrade.
4. Official Nvidia Driver Installation: Finally, I installed the official Nvidia driver for Pop!_OS using sudo apt install system76-driver-nvidia.
5. Reboot: A good old restart (sudo systemctl reboot) was necessary for the changes to take effect.

Reinstall nvidia driver, like what @linuxgnuru says with some adjustment like in this article https://support.system76.com/articles/login-loop-pop/#reinstall-nvidia-driver

```
sudo apt purge ~nnvidia # I change this to `sudo apt purge '^nvidia-*'`
sudo apt autoremove
sudo apt clean
sudo apt update
sudo apt full-upgrade
sudo apt install system76-driver-nvidia
sudo systemctl reboot
```
now I can use my second monitor.

### Important Note:

While switching back to kernel version 6.6.10 temporarily fixed the Bluetooth, staying on the latest stable kernel (currently 6.5.4) is crucial for security reasons. Luckily, the driver update resolved the Bluetooth issue without needing to downgrade the kernel. So, still at kernel version 6.6.10.

**Set Default Boot Kernel**
If you need to change the default kernel version used when the system boots, this can be done with this kernelstub command after downloading it from either `mainline kernel `or `synaptic `:

`sudo kernelstub -v -k /boot/vmlinuz-5.8.18-63.71-care -i /boot/initrd.img-5.8.18-63.71-care`
In this example, we have set the kernel to a 5.8 version compiled by the support team. Substitute the desired version. To see which kernel versions are installed and determine the exact version numbers run:

`dpkg --list | grep linux-image*`
If a specific kernel version is the suspected source of a bug, try a different version.

**Comment by AvryG99** [comment]
I faced the same issue and fixed it but ridiculously didn't know how. I tried many ways on the internet to check on my drivers, services, firmware, BIOS option to even BIOS update. Thought my Bluetooth chip physically died on the motherboard so I booted my machine to a trial Ubuntu via my USB from the BIOS. Well, still no bluetooth on a freshly built OS so I were so sure about it's about a hardware issue. Kinda panic and were gonna buy an external bluetooth usb, but when I booted back to the current ubuntu, the bluetooth somehow appear here. I don't even know what happened here haha
